src/commands/NewIntegrationFileCommand.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as kamel from './../kamel';
+
+const validFilename = require('valid-filename');
+
+const LANGUAGES = ['Java', 'xml', 'Yaml', 'Groovy', 'JavaScript', 'Kotlin'];
+
+export async function create() {
+	let language = await vscode.window.showQuickPick(LANGUAGES);
+	if (language) {
+		let workspaceFolder = await vscode.window.showWorkspaceFolderPick();
+		if (workspaceFolder) {
+			let filename = await vscode.window.showInputBox({
+				prompt: 'Please provide a name for the new file (without extension)',
+				validateInput: (name: string) => {
+					return validateFileName(name, language as string, workspaceFolder as vscode.WorkspaceFolder);
+				}
+			});
+			if (filename) {
+				let kamelExe = kamel.create();
```

Comment:
What is kamelExe? Is it the executable or is it a path to the executable? also kamelExe is never changed again, so it could be a const instead.

Reply:
it is the interface provided by this extension to the Kamel executable

Reply:
I find this name is a bit misleading.

Reply:
ok.
For this PR, i think we need to keep it as-is to stay coherent with other parts of the codebase https://github.com/camel-tooling/vscode-camelk/search?q=kamelExe&unscoped_q=kamelExe
I created https://issues.redhat.com/browse/FUSETOOLS2-348